content/docs/iac/using-pulumi/extending-pulumi/build-a-component.md

```diff
@@ -0,0 +1,2375 @@
+      <version>1.7.36</version>
+    </dependency>
+    <dependency>
+      <groupId>com.google.code.gson</groupId>
+      <artifactId>gson</artifactId>
+      <version>2.8.9</version>
+    </dependency>
+    <dependency>
+      <groupId>com.pulumi</groupId>
+      <artifactId>pulumi</artifactId>
+      <version>[1.6,)</version>
+    </dependency>
+    <dependency>
+      <groupId>com.pulumi</groupId>
+      <artifactId>aws</artifactId>
+      <version>(6.0.2,6.99]</version>
+    </dependency>
+  </dependencies>
+
+  <build>
+    <plugins>
+      <plugin>
+        <groupId>org.codehaus.mojo</groupId>
+        <artifactId>exec-maven-plugin</artifactId>
+        <version>3.0.0</version>
+        <configuration>
+          <mainClass>${mainClass}</mainClass>
+          <commandlineArgs>${mainArgs}</commandlineArgs>
+        </configuration>
+      </plugin>
+    </plugins>
+  </build>
+</project>
+```
+
+The `com.pulumi.pulumi` SDK contains everything we need for making a component. It should be version `1.6` or newer. The `com.pulumi.aws` package is the AWS provider that we are building on top of. We've also included a couple helper libraries like `gson` and `slf4j-nop` which are helpful for this example.
+
+{{% /choosable %}}
+
+{{% choosable language yaml %}}
+
+#### PulumiPlugin.yaml
+
+The `PulumiPlugin.yaml` file tells Pulumi that this directory is a component, rather than a Pulumi program. It's also where we will define the component implementation later. For now, just add the `runtime` and `name` properties.
+
+***Example:** `PulumiPlugin.yaml` for YAML*
+
+```yaml
+runtime: yaml
+name: static-page-component
+```
+
+{{% /choosable %}}
+
+{{< /chooser >}}
+
+#### Define the entrypoint
+
+{{< chooser language "javascript,typescript,python,go,csharp,java,yaml" >}}
+
+{{% choosable language javascript %}}
+
+Authoring sharable components in JavaScript is not currently supported. Considering writing in TypeScript instead!
+
+{{% /choosable %}}
+
+{{% choosable language typescript %}}
+
+First, create the `index.ts` file, where we will export the component class.
+
+***Example:** `index.ts` component export*
+
+```typescript
+export { StaticPage } from "./staticpage";
+```
+
+{{% /choosable %}}
+
+{{% choosable language python %}}
+
+First, create the `__main__.py` file, where we will define an entry point for the component.
+
+***Example:** `__main__.py` component entry point*
+
+```python
+from pulumi.provider.experimental import Metadata, component_provider_host
+from staticpage import StaticPage
+
+if __name__ == "__main__":
+    component_provider_host(metadata=Metadata("static-page-component"), components=[StaticPage])
+```
+
+Here, the `component_provider_host` call invokes a Pulumi provider implmentation which acts as a shim for the component. The `Metadata` we pass to it specifies the name of the component. This name will be important later on in the component implementation, so make sure it's something unique and descriptive!
```

Comment:
```suggestion
Here, the `component_provider_host` call invokes a Pulumi provider implmentation which acts as a shim for the component. The name we pass to it will be important later on in the component implementation, so make sure it's something unique and descriptive!
```